packages/pods-core/src/builtin/GithubPod.ts

```diff
@@ -0,0 +1,318 @@
+
+const ID = "dendron.github";
+
+type GithubImportPodCustomOpts = {
+  owner: string;
+  repository: string;
+  to?: string;
+  from?: string;
+  status: string;
+  token: string;
+  fname: string;
+};
+
+type GithubAPIOpts = GithubImportPodCustomOpts & {
+  created?: string;
+  afterCursor?: string;
+};
+
+type GithubImportPodConfig = ImportPodConfig & GithubImportPodCustomOpts;
+
+export type GithubImportPodPlantOpts = ImportPodPlantOpts;
+
+export class GithubImportPod extends ImportPod<GithubImportPodConfig> {
+  static id: string = ID;
+  static description: string = "import github issues";
+
+  get config(): JSONSchemaType<GithubImportPodConfig> {
+    return PodUtils.createImportConfig({
+      required: ["repository", "owner", "status", "token", "fname"],
+      properties: {
+        owner: {
+          type: "string",
+          description: "owner of the repository",
+        },
+        repository: {
+          description: "github repository to import from",
+          type: "string",
+        },
+        status: {
+          type: "string",
+          description: "status of issue open/closed",
+          enum: ["open", "closed"],
+        },
+        to: {
+          type: "string",
+          description: "import issues created before this date: YYYY-MM-DD",
+          format: "date",
+          default: Time.now().toISODate(),
+          nullable: true,
+        },
+        from: {
+          type: "string",
+          description: "import issues created after this date: YYYY-MM-DD",
+          format: "date",
+          nullable: true,
+        },
+        token: {
+          type: "string",
+          description: "github personal access token",
+        },
+        fname: {
+          type: "string",
+          description: "name of hierarchy to import into",
+        },
+      },
+    }) as JSONSchemaType<GithubImportPodConfig>;
+  }
+
+  /*
+   * method to fetch issues from github graphql api
+   */
+  getDataFromGithub = async (opts: Partial<GithubAPIOpts>) => {
+    let result;
+    const { owner, repository, status, created, afterCursor, token } = opts;
+    let queryVal = `repo:${owner}/${repository} is:issue is:${status}`;
+    if (!_.isUndefined(created)) {
+      queryVal = queryVal.concat(` ${created}`);
+    }
+
+    const query = `query search($val: String!, $after: String)
+    {search(type: ISSUE, first: 100, query: $val, after: $after) {
+      pageInfo {
+        hasNextPage
+        endCursor
+      }
+        edges {
+          node {
+            ... on Issue {
+              title
+              url
+              number
+              state
+              labels(first:5) {
+                edges {
+                  node {
+                    name
+                  }
+                }
+              }
+              body
+            }
+          }
+        }
+      }
+    }`;
+    try {
+      result = await graphql(query, {
+        headers: { authorization: `token ${token}` },
```

Comment:
Is an auth token required even for a public repository?  If it's not required, maybe better to make it an optional param.

Reply:
Yes the token is required since unlike the rest API, GitHub only allows requests at GraphQL API when it's has a token for authentication.